04-onion-routing.md

```diff
@@ -336,9 +341,9 @@ Should this not be the case the node MUST abort processing the packet and report
 
 The node then computes the shared secret as described below, using the private key corresponding to its public key and the ephemeral key from the packet.
 
-The node MUST detect a duplicated routing info which it has already forwarded or redeemed locally; it MAY immediately redeem the HTLC using the preimage (if known), otherwise it MUST abort processing and report a route failure.  This prevents a node on the route from retrying a payment multiple times and attempting to track its progress by traffic analysis.  Note that this could be done using a log of previous shared secrets or HMACs, which can be forgotten once that HTLC would not be accepted anyway (eg. once `outgoing_cltv_value` has passed).  Such a log may use a probabilistic data structure, but MUST rate-limit commitments as necessary to constrain the worst-case storage requirements or false positives of this log.
+The node MUST detect a duplicated routing info which it has already forwarded or redeemed locally; it MAY immediately redeem the HTLC using the preimage (if known), otherwise it MUST abort processing and report a route failure. This prevents a node on the route from retrying a payment multiple times and attempting to track its progress by traffic analysis. Note that this could be done using a log of previous shared secrets or HMACs, which can be forgotten once that HTLC would not be accepted anyway (eg. once `outgoing_cltv_value` has passed). Such a log may use a probabilistic data structure, but MUST rate-limit commitments as necessary to constrain the worst-case storage requirements or false positives of this log.
```

Comment:
The "a" feels inaccurate here in "a duplicated routing info", unless this is genuinely referring to a single packet of information, in which case some singular noun like "packet" should be used.

Reply:
Agreed.  "... a duplicated `onion_packet`" is what we want here.

Reply:
This was fixed in an old commit I think